src/npctalk_funcs.cpp

```diff
@@ -587,6 +587,8 @@ void talk_function::give_equipment_allowance( npc &p, int allowance )
 {
     std::vector<item_pricing> giving = npc_trading::init_selling( p );
     int chosen = -1;
+    // Add asked_for_item effect whether or not NPC has items to give
+    p.add_effect( effect_asked_for_item, 3_hours );
```

Comment:
Couldn't this be done with regular JSON instead of C++? It would be easier that way and more accessible to the general programming base.

Reply:
It can be done via JSON however I'm not sure what else depends on this effect to be added when this function is called. Therefore I could just let the npctalk_funcs.cpp remain the same and modify the JSON to add the effect or just slightly modify this file and wait for someone else to move all effects to the JSON instead of hardecoding it.

Reply:
All you'd need to do is remove the C++ function, make the effect have a time on it, and use an if conditional statement for the associated dialogue so it wouldn't appear if the NPC had this effect.

Reply:
Hi reverted the change in code and made the effect via JSON it so that when we receive item from NPC the effect is added. I did not remove the function entirely because various other functions depend on calling this function and the effect to be added and moving all of them to JSON is out of scope of this PR and I'm not entirely certain what all functions do call it, but perhaps something to look into at a later date.